src/Lib.P2P.Tests/Transports/UdpTest.cs

```diff
@@ -0,0 +1,174 @@
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Lib.P2P.Transports;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests.Transports
+{
+    [TestClass]
+    public class UdpTest
+    {
+        [TestMethod]
+        public void Connect_Missing_UDP_Port()
+        {
+            var udp = new Udp();
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = udp.ConnectAsync("/ip4/127.0.0.1/tcp/32700").Result;
+            });
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = udp.ConnectAsync("/ip4/127.0.0.1").Result;
+            });
+        }
+
+        [TestMethod]
+        public void Connect_Missing_IP_Address()
+        {
+            var udp = new Udp();
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = udp.ConnectAsync("/udp/32700").Result;
+            });
+        }
+
+        [TestMethod]
+        public void Connect_Cancelled()
+        {
+            var udp = new Udp();
+            var cs = new CancellationTokenSource();
+            cs.Cancel();
+            ExceptionAssert.Throws<OperationCanceledException>(() =>
+            {
+                var _ = udp.ConnectAsync("/ip4/127.0.10.10/udp/32700", cs.Token).Result;
+            });
+        }
+
+        [TestMethod]
+        [Ignore]
+        public async Task Listen()
+        {
+            var udp = new Udp();
+            var cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            var connected = false;
+
+            void Handler(Stream stream, MultiAddress local, MultiAddress remote)
+            {
+                Assert.IsNotNull(stream);
+                connected = true;
+            }
+
+            try
+            {
+                var listenerAddress = udp.Listen("/ip4/127.0.0.1", Handler, cs.Token);
+                Assert.IsTrue(listenerAddress.Protocols.Any(p => p.Name == "udp"));
+                await using (var stream = await udp.ConnectAsync(listenerAddress, cs.Token))
+                {
+                    await Task.Delay(50, cs.Token);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)